drivers/display/display_mcux_dcnano_lcdif.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_dcnano_lcdif
+
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+
+#include <fsl_lcdif.h>
+#ifdef CONFIG_HAS_MCUX_CACHE
+#include <fsl_cache.h>
+#endif
+
+
+LOG_MODULE_REGISTER(display_mcux_dcnano_lcdif, CONFIG_DISPLAY_LOG_LEVEL);
+
+#ifdef CONFIG_MCUX_DCNANO_LCDIF_DOUBLE_FRAMEBUFFER
+#define MCUX_DCNANO_LCDIF_FB_NUM 2
+#else
+#define MCUX_DCNANO_LCDIF_FB_NUM 1
+#endif
+
+struct mcux_dcnano_lcdif_config {
+	LCDIF_Type *base;
+	void (*irq_config_func)(const struct device *dev);
+	const struct gpio_dt_spec backlight_gpio;
+	lcdif_dpi_config_t dpi_config;
+	/* Pointer to start of first framebuffer */
+	uint8_t *fb_ptr;
+	/* Number of bytes used for each framebuffer */
+	uint32_t fb_bytes;
+};
+
+struct mcux_dcnano_lcdif_data {
+	uint8_t *fb[MCUX_DCNANO_LCDIF_FB_NUM];
+	uint8_t fb_idx;
+	lcdif_fb_config_t fb_config;
+	uint8_t pixel_bytes;
+	struct k_sem sem;
+};
+
+static int mcux_dcnano_lcdif_write(const struct device *dev, const uint16_t x,
+			     const uint16_t y,
+			     const struct display_buffer_descriptor *desc,
+			     const void *buf)
+{
+	const struct mcux_dcnano_lcdif_config *config = dev->config;
+	struct mcux_dcnano_lcdif_data *data = dev->data;
+	uint8_t next_fb_idx = (data->fb_idx + 1) % MCUX_DCNANO_LCDIF_FB_NUM;
+	uint32_t h_idx;
+	const uint8_t *src;
+	uint8_t *dst;
+
+	__ASSERT((data->pixel_bytes * desc->pitch * desc->height) <=
+		desc->buf_size, "Input buffer too small");
+
+	LOG_DBG("W=%d, H=%d @%d,%d", desc->width, desc->height, x, y);
+
+#ifdef CONFIG_MCUX_DCNANO_LCDIF_DOUBLE_FRAMEBUFFER
+	/* If display write is partial, copy current framebuffer to
+	 * queued one. Note- this has a significant
```

Comment:
"Note- this has a significant" looks weird.

Reply:
Indeed it does:) I've fixed that comment to highlight what it was intending to, namely the potential performance implications of double framebuffering